packages/uicore/src/components/TableV2/TableV2.tsx

```diff
@@ -154,31 +157,35 @@ export const TableV2 = <Data extends Record<string, any>>(props: TableProps<Data
           return (
             // eslint-disable-next-line react/jsx-key
             <div
-              {...row.getRowProps()}
               className={cx(
                 css.row,
                 {
                   [css.card]: !props.minimal,
-                  [css.clickable]: !!props.onRowClick,
-                  [css.minimal]: !!props.minimal
+                  [css.minimal]: !!props.minimal,
+                  [css.clickable]: !!props.onRowClick
                 },
                 getRowClassName?.(row)
-              )}
-              onClick={() => {
-                props.onRowClick?.(row.original, row.index)
-              }}
-              data-testid={rowDataTestID?.(row.original, row.index)}>
-              {row.cells.map((cell, index) => {
-                return (
-                  // eslint-disable-next-line react/jsx-key
-                  <div
-                    {...cell.getCellProps()}
-                    className={css.cell}
-                    style={{ width: headerGroups[0].headers[index].width }}>
-                    {cell.render('Cell')}
-                  </div>
-                )
-              })}
+              )}>
+              <div
+                {...row.getRowProps()}
+                className={css.cells}
+                onClick={() => {
+                  props.onRowClick?.(row.original, row.index)
+                }}
+                data-testid={rowDataTestID?.(row.original, row.index)}>
+                {row.cells.map((cell, index) => {
+                  return (
+                    // eslint-disable-next-line react/jsx-key
+                    <div
+                      {...cell.getCellProps()}
+                      className={css.cell}
+                      style={{ width: headerGroups[0].headers[index].width }}>
+                      {cell.render('Cell')}
+                    </div>
+                  )
+                })}
+              </div>
+              {row.isExpanded && <div className={css.rowSubComponent}>{renderRowSubComponent?.({ row })}</div>}
```

Comment:
this `div` should not be rendered if `renderRowSubComponent` is nullish

Reply:
row.isExpanded would be false in such case